File: src/main/java/cc/mirukuneko/realtrainmodrenewed/rail/util/RailProperties.java
package cc.mirukuneko.realtrainmodrenewed.rail.util;

/**
 * Simplified {@code ResourceStateRail} stand-in: ballast width from rail pack (legacy {@code RailConfig#ballastWidth}).
 */
public class RailProperties {
    /** Same semantics as legacy: full width is2*halfWidth + center; 0 = center column only. */
    public int ballastWidth;
    public float blockHeight;

    public static RailProperties createDefault() {
        RailProperties p = new RailProperties();
        p.ballastWidth = 0;
        p.blockHeight = 0.0625F;
        return p;
    }
}

File: src/main/kotlin/cc/mirukuneko/realtrainmodrenewed/rail/util/RailProperties.kt
package cc.mirukuneko.realtrainmodrenewed.rail.util

/**
 * Simplified ResourceStateRail stand-in: ballast width from rail pack (legacy RailConfig#ballastWidth).
 */
class RailProperties {
    /** Same semantics as legacy: full width is 2*halfWidth + center; 0 = center column only. */
    @JvmField
    var ballastWidth: Int = 0

    @JvmField
    var blockHeight: Float = 0.0625F

    companion object {
        @JvmStatic
        fun createDefault(): RailProperties = RailProperties().apply {
            ballastWidth = 0
            blockHeight = 0.0625F
        }
    }
}
